consensus/safety-rules/src/safety_rules_2chain.rs

```diff
@@ -137,6 +169,16 @@ impl SafetyRules {
         }
     }
 
+    fn safe_to_order_vote(&self, block: &Block) -> Result<(), Error> {
```

Comment:
I don't think we need this check. Any certified block should be safe to sign the order vote. If we check this, we cannot sign order vote for the first block after timeout.